### ISSUE TITLE
Roberta : Training Whole Word Masking with own data [Dutch]

### ISSUE THREAD

**Issue by Skylixia** [pytorch/fairseq#2399]
Hello, 

I would like to train a language model in Dutch using whole word masking. 
I saw this is possible in [this issue](https://github.com/pytorch/fairseq/issues/1576) with --mask-whole-words
For this, I downloaded the Dutch wikipedia dump : nlwiki-20200720-pages-meta-current.xml.bz2 
Then, I ran [WikiExtractor.py](https://github.com/attardi/wikiextractor) to extract and clean the text from this dump. This is done in the same fashin as "[Pre-Training with Whole Word Masking for Chinese BERT](https://www.researchgate.net/publication/333892280_Pre-Training_with_Whole_Word_Masking_for_Chinese_BERT)".
Now I have a folder with alphabetic subfolders containing plain text articles. 
My question is to be sure I follow the right steps to have a whole word masking model, that will predict and tokenize whole words and not word pieces. The goal is to then do fill in mask with the model.
Following "[Pretraining RoBERTa using your own data](https://github.com/pytorch/fairseq/blob/master/examples/roberta/README.pretraining.md)", should I preprocess my text articles by encoding and binarize them with GPT-2 BPE ? Moreover, I imagine I need to pass my own encoder and vocabulary in the parameters : --encoder-json and --vocab-bpe. Would this be all that need to be changed ?
Also, in the documentation the input is a .raw unlike my input. 
Then to do the training should I used the provided command in the [documentation](https://github.com/pytorch/fairseq/blob/master/examples/roberta/README.pretraining.md) with just --mask-whole-words added with True as set value ?

```
TOTAL_UPDATES=125000    # Total number of training steps
WARMUP_UPDATES=10000    # Warmup the learning rate over this many updates
PEAK_LR=0.0005          # Peak learning rate, adjust as needed
TOKENS_PER_SAMPLE=512   # Max sequence length
MAX_POSITIONS=512       # Num. positional embeddings (usually same as above)
MAX_SENTENCES=16        # Number of sequences per batch (batch size)
UPDATE_FREQ=16          # Increase the batch size 16x

DATA_DIR=data-bin/wikitext-103

fairseq-train --fp16 $DATA_DIR \
    --task masked_lm --criterion masked_lm \
    --arch roberta_base --sample-break-mode complete --tokens-per-sample $TOKENS_PER_SAMPLE \
    --optimizer adam --adam-betas '(0.9,0.98)' --adam-eps 1e-6 --clip-norm 0.0 \
    --lr-scheduler polynomial_decay --lr $PEAK_LR --warmup-updates $WARMUP_UPDATES --total-num-update $TOTAL_UPDATES \
    --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.01 \
    --max-sentences $MAX_SENTENCES --update-freq $UPDATE_FREQ \
    --max-update $TOTAL_UPDATES --log-format simple --log-interval 1 \
    --mask-whole-words True
```

Lastly, in --arch roberta_base could this be replaced by another pretrained model on huggingface ? In that case, I could start with robBert, the dutch RoBERTa. 

Thank you for your help and suggestions ! 

Environment
 - OS: Linux
 - How you installed fairseq : git clone + pip install --editable ./
 - Python version: 3.6.9
